test/gpu/native/studies/coral/coral.chpl

```diff
@@ -0,0 +1,199 @@
+
+    Output.localAccess[i,first_point] = (tmpLL + tmpCC + tmpRR + 2*(tmpLC + tmpLR + tmpCR)); // / (Mask_Size**2);
+
+    //var Output = (tmpLL + tmpCC + tmpRR + 2*(tmpLC + tmpLR + tmpCR));
+    var prev = tmpCC + tmpRR + 2*tmpCR;
+
+    for j in (first_point+1)..last_point do {
+
+      tmpLL = 0;
+      tmpLC = 0;
+      tmpLR = 0;
+      tmpCR = 0;
+      tmpRR = 0;
+
+      calc_distance(Array, locL, locL, tmpLL, bs, be, i, j);
+      calc_distance(Array, locL, locC, tmpLC, bs, be, i, j);
+      calc_distance(Array, locL, locR, tmpLR, bs, be, i, j);
+      calc_distance(Array, locC, locR, tmpCR, bs, be, i, j);
+      calc_distance(Array, locR, locR, tmpRR, bs, be, i, j);
+
+      /* Global GPU memory access causing ~20% slowdown? */
+      var current = tmpRR + 2*(tmpLR + tmpCR);
+      Output.localAccess[i,j] = (prev + current); // / (Mask_Size**2);
+
+      //Output = prev + current;
+      prev = prev + current - tmpLL - 2*(tmpLC + tmpLR);
+    }
+
+  }
+
+  if verbose_gpu then stopVerboseGpu();
+
+}
+
+proc main(args: [] string) {
+
+  var t : stopwatch;
+  t.start();
+
+  const radius = (sqrt(window_size) / 2) : int;
+  const nx = (radius / dx) : int;
+  writeln("Distance circle has a radius of ", nx, " points.");
+  writeln("inputSize = ", inputSize);
+
+  var (LeftMask, CenterMask, RightMask, Mask_Size) = create_distance_mask(radius, dx, nx);
+
+  if report_times then
+    writeln("Elapsed time at start of coforall loop: ", t.elapsed(), " seconds.");
+
+  writeln("Starting coforall loop.");
+
+  coforall loc in Locales do on loc {
+
+    on here.gpus[0] {
+
+      const radius = (sqrt(window_size) / 2) : int;
+      const nx = (radius / dx) : int;
+      writeln("Distance circle has a radius of ", nx, " points.");
+
+      var x, y: int;
+      if inputSize != -1 {
+        x = inputSize;
+        y = inputSize;
+      }
+      else if bigInput {
+        x = 15243;
+        y = 10073;
+      }
+      else {
+        x = 1000;
+        y = 1000;
+      }
+      const ImageSpace = {0..#x, 0..#y};
+
+
+      var Array : [1..5,1..y,1..x] real(32);
+
+      // Read in array
+      if inputSize == -1 {
+        var f = open(in_array, ioMode.r);
+        var r = f.reader(kind=ionative);
+        for i in 1..5 {
+          for j in 1..x {
+            for k in 1..y {
+              var tmp : real;
+              r.readBinary(tmp);
+              Array[i,k,j] = tmp : real(32);
+            }
+          }
+        }
+        r.close();
+      }
+      else {
+        use Random;
+        fillRandom(Array, seed=13);
+        if write_data then writeln(Array);
+      }
+
+      // Create Block distribution of interior of PNG
+      const offset = nx+1; // maybe needs to be +1 to account for truncation?
+      const Inner = ImageSpace.expand(-offset);
+      const myTargetLocales = reshape(Locales, {1..Locales.size, 1..1});
+      const D = Inner dmapped Block(Inner, targetLocales=myTargetLocales);
```

Comment:
I don't see this used anywhere -- is this a carry-over from the non-GPU implementation? Should we delete this code, along with the `reshape` operation above?

Reply:
Good point. How about I keep it as commented with some description?

Your assessment is probably correct, but we also want to have a similar thing going forward where we modify this to be multilocale and multi GPU.